calculator/lexer.py

```diff
@@ -11,8 +45,88 @@ def lex(s):
       * "*" - Multiplication
       * "+" - Addition
       * "-" - Subtraction
+      * "/" - Division
     * The following grouping symbols are supported:
       * "("
       * ")"
     """
-    return []
+
+    # Defining char_before and char_after
+    char_before = None
+    char_after = None
+    # Partial_num is the var that stores my nums until I have to append it to my tokens list
+    partial_num = None
+    # List that stores the tokens
+    tokens = []
+    # Stripping the input line of spaces
+    stripped_line = line.replace(" ", "")
+
+    # Iterating through each index in my stripped_line
+    for i in range(0, len(stripped_line)):
+        # Defining "c" as the char at the index "i" in stripped_line
+        c = stripped_line[i]
+        # Defining char_before but if i is the first index then the index will be invalid
+        if i != 0:
+            # Char_before is the current char's index (i), but subtracting 1 because it is 1 before
+            char_before = stripped_line[i - 1]
+        else:
+            # if i is the first index then char_before is None
+            char_before = None
+
+        # Catching the same error as the first if statement but if i is the last index
+        if i != len(stripped_line) - 1:
+            # Char_after is the current char's index (i), but adding 1 because it is 1 after
+            char_after = stripped_line[i + 1]
+        else:
+            # If i is the last index then char_after is None
+            char_after = None
+        # If c (current char), is a digit or a decimal point. If c is a int or a float:
+        if c.isdigit() or c == ".":
+            # If c is the first iteration:
+            if partial_num == None:
+                # Partial_num is equal to c instead of partial_num is the original value but c added on.
+                partial_num = str(c)
+                # adding this char to partial num
+            else:
+                partial_num += str(c)
+            # If the char_after is a operator or a parenthesis:
+            if is_op(char_after) == True or is_parenthisis(char_after) == True:
+                # If this condition is true then we know that the number or decimal is done.
+                tokens.append(convert(partial_num))
+                # Then we have to reset partial_num.
+                partial_num = None
+            # We do the same as the above if statement, if it is the end of the line.
+            elif i == len(stripped_line) - 1:
+                tokens.append(convert(partial_num))
+                partial_num = None
+
+        # If c is a negative sign or a subtraction operator:
+        elif c == "-":
+            # If it is the first char in the line:
+            if stripped_line[0] == c:
```

Comment:
This does not do what your comment indicates.  Can you figure out why?

Reply:
Yes, I have to check the index not the character! 👍